### PR TITLE
Added '0x' behaviour + removal of lodash on filters

### PR DESCRIPTION
**Pull request by CP2053**
No text.

### CODE REVIEW

File: reactComponents/src/searchbar/redux/tokenSearchSlice.ts [hunk 1]
```diff
@@ -25,91 +26,56 @@ export const resetSearchOnNewExchange = createAsyncThunk(
...
       // query an unused echange for a given network.
-      processedNetworks = processedNetworks
+      networkMap = networkMap
         .filter(network => networkExchangePairs
-          .filter(pair => pair[0] === network && processedExchanges.includes(pair[1])).length >= 1);
+          .filter(pair => pair[0] === network && exchangeMap.includes(pair[1])).length >= 1);
 
       // Loading the data.
-      const data = await retry(() => searchTokensAsync(searchString, processedNetworks, processedExchanges), { retries: 1 });
+      const data = await retry(() => searchTokensAsync(searchString, networkMap, exchangeMap), { retries: 1 });
```
Comment (iandjx) [inline reactComponents/src/searchbar/redux/tokenSearchSlice.ts:75]: why do we need to retry?

File: reactComponents/src/searchbar/redux/tokenSearchSlice.ts [hunk 2]
```diff
@@ -25,91 +26,56 @@ export const resetSearchOnNewExchange = createAsyncThunk(
...
 
       // console.log("data", data);
-      console.log("data", data.length);
+      console.log(data.length + ' results', (new Date().getTime() - pairSearchTimestamp) + 'ms');
```
Comment (iandjx) [inline reactComponents/src/searchbar/redux/tokenSearchSlice.ts:78]: remove console.log

File: reactComponents/src/searchbar/tokenSearch/SearchFilters.tsx [hunk 3]
```diff
@@ -28,7 +40,11 @@ export const SearchFilters = () => {
         <AccordionItemHeading>
           <AccordionItemButton >
             <div tw="p-4 flex">
-              <div tw="font-bold">Filter Networks:</div>  &nbsp; Searching {networkCount} networks and {exchangeCount} exchanges
+              <div tw="font-bold">Filter Networks:</div>&nbsp;Searching {
```
Comment (iandjx) [inline reactComponents/src/searchbar/tokenSearch/SearchFilters.tsx:43]: https://github.com/Rome-Blockchain-Labs/terminal-library/pull/12 doesnt use tailwind anymore

**Review by iandjx — CHANGES_REQUESTED** [review]

File: reactComponents/dist/index.mjs [hunk 4]
```diff
@@ -182,35 +185,22 @@ var resetSearchOnNewExchange = createAsyncThunk("token/searchReset", async (sear
...
-};
 var searchTokenPairs = createAsyncThunk("token/search", async (searchString, thunkAPI) => {
+  if (!searchString)
+    return;
   try {
     let { networkMap, exchangeMap } = thunkAPI.getState();
-    let processedNetworks;
-    let processedExchanges;
     const pairSearchTimestamp = new Date().getTime();
+    console.log("Search sent");
```
[the remark this answers is not in the document]
Comment (oneway-fullstack) [reply reactComponents/dist/index.mjs:194]: Ok, no problem

File: reactComponents/src/searchbar/redux/tokenSearchSlice.ts (hunk 1 above)
Comment (oneway-fullstack) [reply reactComponents/src/searchbar/redux/tokenSearchSlice.ts:75]: We need it, it's to get the token pair by network and exchange, although it's not what I wrote.

File: reactComponents/src/searchbar/redux/tokenSearchSlice.ts (hunk 2 above)
Comment (oneway-fullstack) [reply reactComponents/src/searchbar/redux/tokenSearchSlice.ts:78]: Ok

File: reactComponents/src/searchbar/tokenSearch/SearchFilters.tsx (hunk 3 above)
Comment (oneway-fullstack) [reply reactComponents/src/searchbar/tokenSearch/SearchFilters.tsx:43]: Which branch do you review now?
I only work on search-bar branch.

**Pull request closed without merging**